osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -226,10 +231,13 @@ protected sealed override void ApplyResultInternal(JudgementResult result)
 
             ApplyScoreChange(result);
 
-            hitEvents.Add(CreateHitEvent(result));
-            lastHitObject = result.HitObject;
+            if (!IsSimulating && TrackHitEvents)
```

Comment:
This feels like a bit of a much of muchness? `IsSimulating` and `TrackHitEvents` are basically the same thing. So why not just keep `IsSimulating` from the base, make it `internal`, and do away with `TrackHitEvents` entirely?

Reply:
See my comments in this PR regarding optimisation. The first is applied to normal gameplay / processing and is a pretty large saving. The second is further optimisation specifically for this PR.

Reply:
I get what each is doing, just saying it feels a bit superfluous to have 2 flags steering basically the same thing rather than just one generic one for "simulation" that implementations use to skip processing that isn't necessary in actual gameplay contexts. Doesn't matter much though.